test/cmds/manifest/manifest.test.js

```diff
@@ -0,0 +1,318 @@
+            url: 'http://localhost:8081/dist/bundle.js',
+          },
+          initialPermissions: { alert: 'foo', read: ['foo', 'bar'] },
+        }));
+      await expect(manifest(mockArgv)).rejects.toThrow('Manifest Error: package.json validation failed, please see above errors.');
+      expect(global.console.error).toHaveBeenCalledTimes(2);
+    });
+
+    it('checks web3Wallet initial permissions property: handles valid and throws error if object\'s permission keys are invalid', async () => {
+      jest.spyOn(JSON, 'parse').mockImplementation((value) => value);
+      jest.spyOn(utils, 'isFile').mockImplementation(() => true);
+      jest.spyOn(console, 'error').mockImplementation();
+      jest.spyOn(fs.promises, 'readFile')
+        .mockImplementationOnce(async () => await getPackageJson({
+          bundle: {
+            local: 'dist/bundle.js',
+            url: 'http://localhost:8081/dist/bundle.js',
+          },
+          initialPermissions: { approve: { invoker: 'foo', date: 4546 }, bad: { dangerous: 'scary' }, parent: { parentCapability: 'mother' } },
+        }));
+      await expect(manifest(mockArgv)).rejects.toThrow('Manifest Error: package.json validation failed, please see above errors.');
+      expect(global.console.error).toHaveBeenCalledTimes(2);
+    });
+
+    describe('checks if log error and warning aligns to global settings', () => {
+
+      afterEach(() => {
+        global.snaps.verboseErrors = false;
+        global.snaps.suppressWarnings = false;
+      });
+
+      it('checks logManifestError prints according to global settings', async () => {
+        global.snaps.verboseErrors = true;
+        jest.spyOn(JSON, 'parse').mockImplementation((value) => value);
+        jest.spyOn(utils, 'isFile').mockImplementation(() => true);
+        jest.spyOn(console, 'error').mockImplementation();
+        jest.spyOn(fs.promises, 'readFile')
+          .mockImplementationOnce(async () => await getPackageJson({
+            bundle: {
+              local: 'dist/bundle.js',
+              url: 'http://localhost:8081/dist/bundle.js',
+            },
+            initialPermissions: { alert: 'foo', read: ['foo', 'bar'] },
+          }));
+        await expect(manifest(mockArgv)).rejects.toThrow('Manifest Error: package.json validation failed, please see above errors.');
+        expect(global.console.error).toHaveBeenCalledTimes(2);
+      });
+
+      it('checks logManifestWarning prints according to global settings', async () => {
+        global.snaps.suppressWarnings = true;
+        const badJSON = {};
+        jest.spyOn(JSON, 'parse').mockImplementation((value) => value);
+        jest.spyOn(console, 'warn').mockImplementation();
+        jest.spyOn(console, 'log').mockImplementation();
+        jest.spyOn(fs.promises, 'readFile')
+          .mockImplementationOnce(async () => badJSON);
+        await manifest(mockArgv);
+        expect(global.console.warn).not.toHaveBeenCalled();
+      });
```

Comment:
The name of this test is somewhat unclear to me. Is it checking whether manifest warnings are suppressed? If so, a better name may be: `suppresses logManifestWarning per global settings`